From c23e9d1e05f045b0bfaa435068de68a6340982e2 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Sun, 9 Nov 2025 21:02:43 +0000
Subject: [PATCH] sshd-auth.c: guard call to OpenSSL_add_all_algorithms

In OpenSSL OpenSSL_add_all_algorithms is deprecated, and can be
missing when OpenSSL is compiled without deprecated functionality.

There is existing check for presence of this function or macro, thus
guard the call to it base on the configure time check.
---
 sshd-auth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sshd-auth.c b/sshd-auth.c
index 9c31515de..acfecbdc8 100644
--- a/sshd-auth.c
+++ b/sshd-auth.c
@@ -588,7 +588,7 @@ main(int ac, char **av)
 	if (!rexeced_flag)
 		fatal("sshd-auth should not be executed directly");
 
-#ifdef WITH_OPENSSL
+#if defined(WITH_OPENSSL) && defined(HAVE_OPENSSL_ADD_ALL_ALGORITHMS)
 	OpenSSL_add_all_algorithms();
 #endif
 
-- 
2.51.0

